### ISSUE TITLE
"master" toolchain no longer accepts -Z flags 

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust#79160]
On nightly (`rustc 1.50.0-nightly (c919f490b 2020-11-17`) I can run 
`/home/matthias/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/rustc ./75323.rs  -Zsave-analysis` which works fine, but this no longer works with the master toolchain (c4f836ad1aceb83507810d9499f56988fd24578d); this worked yesterday!
`/home/matthias/.rustup/toolchains/master/bin/rustc ./75323.rs  -Zsave-analysis`
```
error: the option `Z` is only accepted on the nightly compiler
```

**Comment by matthiaskrgr** [comment]
Just saw https://github.com/rust-lang/rust/pull/79159  , could be related

**Comment by pietroalbini** [comment]
This is definitely #79159, sorry about that.

**Comment by Stupremee** [comment]
Assigning `P-critical` as [discussed as part of the Prioritization Working Group procedure](https://rust-lang.zulipchat.com/#narrow/stream/245100-t-compiler.2Fwg-prioritization.2Falerts/topic/I-prioritize.20.2379160.20.22master.22.20toolchain.20no.20longer.20accepts.20-Z.E2.80.A6/near/217122153) and removing `I-prioritize`.

**Comment by matthiaskrgr** [comment]
Fixed by #79159
